nilearn/_utils/niimg.py

```diff
@@ -0,0 +1,157 @@
+"""
+Neuroimaging file input and output.
+"""
+# Author: Gael Varoquaux, Alexandre Abraham, Philippe Gervais
+# License: simplified BSD
+
+import copy
+import gc
+import collections
+from six import string_types
+
+import numpy as np
+import nibabel
+
+from .numpy_conversions import as_ndarray
+
+
+def _safe_get_data(img):
+    """ Get the data in the image without having a side effect on the
```

Comment:
> Does anybody know what problem solves this function? From what I understand, if
> the data is memmaped, it loads it into memory. First:

I think that the goal is to prevent memory leaks: when 'get_data' is
called, the niftiimage object has the data attached to it. If a reference
is kept on this object, outside of the function, we have a memory leak,
as the function just loaded data that won't be removed.